Tests/WebApi.UnitTests/Applications/DirectorOperations/Commands/CreateDirector/CreateDirectorCommandValidatorTest.cs

```diff
@@ -0,0 +1,42 @@
+using AutoMapper;
+using FluentAssertions;
+using MovieStoreWebApi.Applications.ActorOperations.Commands.CreateActor;
+using MovieStoreWebApi.Applications.DirectorOperations.Commands.CreateDirector;
+using MovieStoreWebApi.DbOperations;
+using Tests.WebApi.UnitTests.TestSetup;
+using Xunit;
+
+namespace Tests.WebApi.UnitTests.Applications.DirectorOperations.Commands.CreateActor
+{
+    public class CreateDirectorCommandValidatorTest : IClassFixture<CommonTestFixture>
+    {
+        private readonly MovieStoreDbContext _context;
+        private readonly IMapper _mapper;
+
+        public CreateDirectorCommandValidatorTest(CommonTestFixture testFixture)
+        {
+            _context = testFixture.Context;
+            _mapper = testFixture.Mapper;
+        }
+
+        [InlineData("Christopher", "Nolan")]
+        [InlineData("William", "")]
+        [InlineData("Fatih", "Yapici")]
+
+        [Theory]
+        public void WhenInvalidInputsAreGiven_InvalidOperationException_ShouldReturn(string name, string surname)
+        {
+            CreateDirectorCommand command = new CreateDirectorCommand(null, null);
+            command.Model = new CreateDirectorModel()
+            {
+                Name = name,
+                Surname = surname,
+            };
+
+            CreateDirectorCommandValidator validator = new CreateDirectorCommandValidator();
+            var result = validator.Validate(command);
+
+            result.Errors.Count.Should().BeGreaterThan(0);
```

Comment:
Burada metodun ismi dogrudan InvalidOperationException donmesi gerektigini belirtiyor mesela. Burada bu exception turunde donus oldugunu denetlemen gerekmiyor mu?

Reply:
Hayır. CreateDirectorCommandValidator içerisinde birçok farklı kuraldan geçeceği için belirli bir hata mesajı verilmesine gerek yok. Sadece verilen inputların belirli kurallara uygun olup olmadığı yeterli. Exception türünde dönüşler CreateDirectorCommand içerisinde mevcut.

Reply:
InvalidOperationException hata mesaji degil hatanin kendisi direkt. Bu cevap beni cok tatmin etmedi ama kucuk bir detay zaten, bu sekilde kullaniliyorsa sorun degil.